.github/workflows/conda_unit_test.yml

```
@@ -1,24 +1,76 @@
 name: Unit Tests
 
-on: [push, pull_request]
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
+  workflow_dispatch:   # allow manual triggering
+
+defaults:
+  run:
+    shell: bash -l {0}
 
 jobs:
 
+  lint:
+    name: Code style
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v3
```

Version 4 is the latest now.
```suggestion
      - uses: actions/checkout@v4
```